mobile_verifier/src/reward_shares.rs

```diff
@@ -183,9 +288,90 @@ mod test {
         speedtests::{Speedtest, SpeedtestAverages},
     };
     use chrono::{Duration, NaiveDateTime, Utc};
+    use futures::stream;
     use helium_proto::services::poc_mobile::HeartbeatValidity;
     use std::collections::{HashMap, VecDeque};
 
+    #[tokio::test]
+    async fn transfer_reward_amount() {
+        let owner: PublicKeyBinary = "112NqN2WWMwtK29PMzRby62fDydBJfsCLkCAf392stdok48ovNT6"
+            .parse()
+            .expect("failed owner parse");
+        let payer: PublicKeyBinary = "11sctWiP9r5wDJVuDe1Th4XSL2vaawaLLSQF8f8iokAoMAJHxqp"
+            .parse()
+            .expect("failed payer parse");
+
+        let data_transfer_sessions = stream::iter(vec![ValidDataTransferSession {
+            pub_key: owner.clone(),
+            payer,
+            upload_bytes: 66,
+            download_bytes: 1,
+            num_dcs: 0, // Not used
+            first_timestamp: DateTime::default(),
+            last_timestamp: DateTime::default(),
+        }]);
+
+        let now = Utc::now();
+        let epoch = (now - Duration::hours(1))..now;
+
+        let data_transfer_rewards =
+            TransferRewards::from_transfer_sessions(dec!(1.0), data_transfer_sessions, &epoch)
+                .await
+                .expect("Could not fetch data transfer sessions");
+
+        assert_eq!(data_transfer_rewards.reward(&owner), dec!(0.00000006));
+        assert_eq!(data_transfer_rewards.scale, dec!(1.0));
+        // Is this correct?
+        assert_eq!(
+            data_transfer_rewards.remaining_rewards,
+            dec!(4166666666666.6666666066666666)
+        );
+    }
+
+    #[tokio::test]
+    async fn transfer_reward_scale() {
+        let owner: PublicKeyBinary = "112NqN2WWMwtK29PMzRby62fDydBJfsCLkCAf392stdok48ovNT6"
+            .parse()
+            .expect("failed owner parse");
+        let payer: PublicKeyBinary = "11sctWiP9r5wDJVuDe1Th4XSL2vaawaLLSQF8f8iokAoMAJHxqp"
+            .parse()
+            .expect("failed payer parse");
+
+        // Just an absurdly large amount of DC
+        let mut transfer_sessions = Vec::new();
+        for _ in 0..1_000_000 {
+            transfer_sessions.push(ValidDataTransferSession {
+                pub_key: owner.clone(),
+                payer: payer.clone(),
+                upload_bytes: 66 * 80_000_000 * 95_000_000,
+                download_bytes: 0,
+                num_dcs: 0, // Not used
+                first_timestamp: DateTime::default(),
+                last_timestamp: DateTime::default(),
+            });
+        }
+
+        let data_transfer_sessions = stream::iter(transfer_sessions);
+
+        let now = Utc::now();
+        let epoch = (now - Duration::hours(24))..now;
+
+        let data_transfer_rewards =
+            TransferRewards::from_transfer_sessions(dec!(1.0), data_transfer_sessions, &epoch)
+                .await
+                .expect("Could not fetch data transfer sessions");
+
+        assert_eq!(
```

Comment:
difficult to understand the expected values here.  Maybe add a comment detailing how those are arrived at

Reply:
i'm not sure how we get these numbers; this test establishes the epoch to be a 24 hour period and the test above establishes the _total_ rewards for a 24 period are `dec!(100_000_000)` so how are we getting these significantly larger numbers? and yes, a comment to indicate 33_333_333_333_333 is expected to be 20% of total rewards for a 24 hour period and 66_666_666_666_666 is 40% of the total rewards for a 24 hour period